jni/include/faiss_stream_support.h

```diff
@@ -56,6 +56,40 @@ class FaissOpenSearchIOReader final : public faiss::IOReader {
 };  // class FaissOpenSearchIOReader
 
 
+/**
+ * A glue component inheriting IOWriter to delegate IO processing down to the given
+ * mediator. The mediator is expected to do write bytes via the provided Lucene's IndexOutput.
+ */
+class FaissOpenSearchIOWriter final : public faiss::IOWriter {
+ public:
+  explicit FaissOpenSearchIOWriter(NativeEngineIndexOutputMediator *_mediator)
```

Comment:
If NativeEngineIndexOutputMediator is managed by this class (i.e., it should be deleted when FaissOpenSearchIOWriter is destroyed), Could we consider using a std::unique_ptr instead of a raw pointer to ensure proper cleanup?

Reply:
We can avoid heap (I'm referring the heap in OS memory layout, not java) memory by creating the object in stack area.
I chose this way in purpose to avoid calling malloc, as glic + jmalloc has an internal locking mechanism which probably have an impact on performance.

Reply:
Make Sense